apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/model/ModelBuilder.kt

```diff
@@ -103,4 +112,42 @@ internal class ModelBuilder(
       }
     }
   }
+
+  private fun TypeSpec.addBuilder(context: JavaContext): TypeSpec {
+    val fields = fieldSpecs.filter { !it.modifiers.contains(Modifier.STATIC) }
+      .filterNot { it.name.startsWith(prefix = "$") }
+    if (fields.isEmpty()) {
+      return this
+    } else {
+      val builderVariable = JavaClassNames.Builder.simpleName().decapitalizeFirstLetter()
+      val builderClass = ClassName.get("", JavaClassNames.Builder.simpleName())
+      val toBuilderMethod = MethodSpec.methodBuilder(Builder.TO_BUILDER_METHOD_NAME)
+        .addModifiers(Modifier.PUBLIC)
+        .returns(builderClass)
+        .addStatement("\$T \$L = new \$T()", builderClass, builderVariable, builderClass)
+        .addCode(fields
+          .map { CodeBlock.of("\$L.\$L = \$L;\n", builderVariable, it.name, it.type.unwrapOptionalValue(it.name)) }
```

Comment:
You'll need to escape the property name:

```suggestion
          .map { CodeBlock.of("\$L.\$L = \$L;\n", builderVariable, context.layout.propertyName(it.name), it.type.unwrapOptionalValue(it.name)) }
```